### PR TITLE
Revision index - other APIs

### PR DESCRIPTION
**Pull request by ChGen**
Here's Revision index extra APIs with tests.

### CODE REVIEW

**Comment by ChGen** [comment]
@olenagerasimova , please review.

File: src/test/java/com/artipie/conan/RevisionsIndexApiTest.java [hunk 1]
```diff
@@ -148,6 +150,114 @@ void updateBinaryIndex(final String[] files) {
...
+        MatcherAssert.assertThat("Expect 1 recipe revision", revs.size() == 1);
+        MatcherAssert.assertThat("rev[0] must be zero", revs.get(0) == 0);
+    }
+
+    @Test
+    void getBinaryRevisions() {
+        new TestResource(RevisionsIndexApiTest.DIR_PREFIX).addFilesTo(this.storage, Key.ROOT);
+        final List<Integer> binrevs = this.index.getBinaryRevisions(
+            0, RevisionsIndexApiTest.ZLIB_BIN_PKG
+            ).toCompletableFuture().join();
```
Comment (olenagerasimova) [inline src/test/java/com/artipie/conan/RevisionsIndexApiTest.java:166]: formatting here is off

**Review by olenagerasimova — CHANGES_REQUESTED** [review]
@ChGen thanks, just one minor comment. Also, it seems like `RevisionsIndexApi` class gets bigger and bigger, is it possible to divide it somehow? Cohesion also does not look good, for example, only one new method from 11 uses `RevisionsIndexer` field...

**Comment by ChGen** [comment]
@olenagerasimova , please review.
In the current APIs `FullIndexer` is used once and `RevisionsIndexer` is used twice (for recipes and binaries) + referenced in `FullIndexer`. This is the last subtle set of changes for Revision index part, so my current plan is to keep the API in single interface for now, as is, and in the future refactor both indexers out of the main interface, so it will be smaller and with better cohesion.

**Review by olenagerasimova — APPROVED** [review]
@ChGen ok, thanks

**Pull request merged**